promql/engine.go

```diff
@@ -368,13 +380,15 @@ func (ng *Engine) execEvalStmt(ctx context.Context, query *query, s *EvalStmt) (
 	}
 
 	evalTimer := query.stats.GetTimer(stats.InnerEvalTime).Start()
-	// Instant evaluation.
+	// Instant evaluation. This is executed as a range evaluation with one step.
 	if s.Start == s.End && s.Interval == 0 {
 		start := timeMilliseconds(s.Start)
 		evaluator := &evaluator{
-			Timestamp: start,
-			ctx:       ctx,
-			logger:    ng.logger,
+			Timestamp:    start,
+			EndTimestamp: start,
+			Interval:     1,
```

Comment:
What's the purpose of exporting (capitalizing) all these fields of an unexported struct type?

Reply:
Fixed